hive/src/main/java/com/netease/arctic/hive/op/ReplaceHivePartitions.java

```diff
@@ -221,6 +226,25 @@ private void applyHivePartitions() {
     }
   }
 
+  /**
+   * check files in the partition, and delete orphan files
+   * @param partitionLocation
+   * @param dataFiles
+   */
+  private void checkOrphanFilesAndDelete(String partitionLocation, List<DataFile> dataFiles) {
+    if (!table.io().isEmptyDirectory(partitionLocation)) {
+      List<String> filePathCollect = dataFiles.stream()
+          .map(dataFile -> dataFile.path().toString()).collect(Collectors.toList());
+      List<FileStatus> exisitedFiles = table.io().list(partitionLocation);
+      for (FileStatus filePath: exisitedFiles) {
+        if (!filePathCollect.contains(filePath.getPath().toString())) {
+          table.io().deleteFile(String.valueOf(filePath.getPath().toString()));
```

Comment:
When deleting a file, try to print why it was deleted and the name of the deleted file to the WARN log.